lib/utils.js

```diff
@@ -592,9 +603,42 @@ export function moveElementAwayFromCollision(
         compactType,
         cols
       );
+    } else if (collisionNorth && compactV) {
+      return moveElement(
+        layout,
+        itemToMove,
+        undefined,
+        collidesWith.y + 1,
+        isUserAction,
+        preventCollision,
+        compactType,
+        cols
+      );
+    } else if (collisionNorth && compactType == null) {
```

Comment:
Hello @dseif thanks for this awesome feature, we want to use it but updating the library lead us into a bug.

As we’re using the library is with a compactType={null} and preventCollision={false} that give us a behavior of a canvas where you can place items wherever you	want, and on draging an element it can push others around the canvas, also if there is a static element it must not be pushed. That worked just fine on the version 1.3.0  Here the example running: 

https://github.com/react-grid-layout/react-grid-layout/assets/5472392/a1424cca-4167-47bc-8814-996d0b5489f5

https://codesandbox.io/p/sandbox/heuristic-diffie-d9wg7g

But updating the version to get this feature, we lost the desired behavior, and I realized this else if is causing it, I would like to understand what is this for.

The unexpected behavior is here:


https://github.com/react-grid-layout/react-grid-layout/assets/5472392/cb895f16-9f63-4f13-8a84-e4f3126f3ea1

 https://codesandbox.io/p/sandbox/mutable-paper-r4f6rq 

Reply:
Hi @ManfredA, sorry you're running into this. I would recommend filing an issue to get someone to look into this sooner. Unfortunately I'm busy with some other work projects at the moment, so not sure when I'll have time to take a look into this.

Reply:
At a quick 1 minute glance it looks like `preventCollision` isn't being adhered to for this condition. The others take it into account, but for w.e reason it was missed here. You could try your had at opening a PR and diving into that a bit. Likely warrants a test of its own as well since this was missed.